rest-servlet/src/main/java/org/hawkular/metrics/restServlet/influx/InfluxSeriesHandler.java

```diff
@@ -435,11 +437,65 @@ private String getColumnName(SelectQueryDefinitions queryDefinitions) {
                     NumberFunctionArgument argument = (NumberFunctionArgument) aggregationFunctionArguments.get(1);
                     retVal = quantil(list, argument.getDoubleValue());
                     break;
+                case TOP:
+                    isMultipleValues = true;
+                    if (!list.isEmpty() && aggregationFunctionArguments.size() > 0) {
+                        int numberOfTopElement = list.size() >
+                            Integer.valueOf(aggregationFunctionArguments.get(0).toString()) ?
```

Comment:
The first function argument is the metric name. What you want is the second argument. Besides, you don't need this conversion. Just cast the argument to the appropriate type (the validator should have checked the type earlier):
```java
NumberFunctionArgument argument = (NumberFunctionArgument) aggregationFunctionArguments.get(1);
```

See https://github.com/hawkular/hawkular-metrics/pull/148/files#diff-9ca02f4743984c633709aa4fbd223720L435